internal/expr/bindtypes.go

```diff
@@ -131,29 +100,24 @@ func (pe *ParsedExpr) BindTypes(args ...any) (te *TypedExpr, err error) {
 
 // bindInputTypes binds the input expression to a type and returns the
 // typeMember represented by the expression.
-func bindInputTypes(ti typeNameToInfo, e *inputExpr) (tm typeinfo.Member, err error) {
+func bindInputTypes(argInfo typeinfo.ArgInfo, e *inputExpr) (input typeinfo.Input, err error) {
```

Comment:
If you reorder these arg lists, putting subject ahead of auxiliary data, you can start to see their candidacy for putting on the input/output expression types themselves.

Not advocating for this right now.

Reply:
Agreed, I make this an interface method on `expression` in #112